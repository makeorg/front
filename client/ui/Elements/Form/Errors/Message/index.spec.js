import React from 'react';
import renderer from 'react-test-renderer';
import {
  LoginErrorMessage,
  MessageWithDynamicLabel,
  DefaultApiErrorMessage,
} from './index';

describe('Form Error Messages', () => {
  it('snapshot LoginErrorMessage', () => {
    const component = renderer.create(<LoginErrorMessage />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('snapshot DefaultApiErrorMessage', () => {
    const component = renderer.create(<DefaultApiErrorMessage />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('snapshot MessageWithDynamicLabel', () => {
    const component = renderer
      .create(
        <MessageWithDynamicLabel
          messageKey="foo_message_key"
          field="foo"
          labelKey="foo_label_key"
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
