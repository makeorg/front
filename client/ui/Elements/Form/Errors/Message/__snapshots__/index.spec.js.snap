// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Error Messages snapshot DefaultApiErrorMessage 1`] = `"common.form.messages.api_error"`;

exports[`Form Error Messages snapshot LoginErrorMessage 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "login.email_doesnot_exist",
    }
  }
/>
`;

exports[`Form Error Messages snapshot MessageWithDynamicLabel 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "foo_message_key",
    }
  }
/>
`;
